### PR TITLE
Retry getting the cgroup root at apply time.

### PR DESCRIPTION
**Pull request by vmarmol**
This will allow late-binding of the cgroup hierarchy.

Signed-off-by: Victor Marmol <vmarmol@google.com>

### CODE REVIEW

**Comment by vmarmol** [comment]
@jaybuff

**Comment by LK4D4** [comment]
Maybe we can then remove `init` and just compute root lazyli

**Comment by vmarmol** [comment]
Good idea, done.

**Comment by jaybuff** [comment]
perhaps include in the commit message that this resolves https://github.com/docker/docker/issues/8791

**Comment by vmarmol** [comment]
@jaybuff added it to the commit.

File: cgroups/fs/apply_raw.go [hunk 1]
```diff
@@ -27,18 +27,20 @@ var (
...
 	cpuRoot, err := cgroups.FindCgroupMountpoint("cpu")
 	if err != nil {
-		return
+		return err
 	}
 	cgroupRoot = filepath.Dir(cpuRoot)
 
 	if _, err := os.Stat(cgroupRoot); err != nil {
-		return
+		return err
```
Comment (LK4D4) [inline cgroups/fs/apply_raw.go]: Don't know how it was before, but if `cgroupRoot` is not `""` and `os.Stat` failed, then second time `setCgroupRoot` won't be called. Also this function now can be called from concurrent goroutines.
So, resume:
* we need set cgroupRoot back to `""` on `Stat` fail
* we need to protect it with mutex

Comment (vmarmol) [inline cgroups/fs/apply_raw.go]: Synchronized access and generation of cgroupRoot.

**Comment by LK4D4** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**